### PR TITLE
ci-upgrade: Add e2e upgrade tests

### PR DESCRIPTION
**Pull request by brb**
Successful run - https://github.com/cilium/cilium/actions/runs/6038229236/job/16384162015.

### CODE REVIEW

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+  # renovate: datasource=github-releases depName=cilium/cilium-cli
+  cilium_cli_version: v0.15.5
+  cilium_cli_ci_version:
+  check_url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+  cilium_stable_version: 1.14
+
+jobs:
+  commit-status-start:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (brlbil) [inline .github/workflows/tests-e2e-upgrade.yaml:64]: With recent PR #27391 we now add names to every Jobs and Steps
New PR #27839  adds linting for this not merged yet

\+ name: Commit Status Star

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+
+      - name: Publish Test Results As GitHub Summary
+        if: ${{ always() }}
+        uses: aanm/junit2md@332ebf0fddd34e91b03a832cfafaa826306558f9 # v0.0.3
+        with:
+          junit-directory: "cilium-junits"
+
+  commit-status-final:
+    if: ${{ always() }}
+    needs: setup-and-test
```
Comment (brlbil) [inline .github/workflows/tests-e2e-upgrade.yaml:431]: Same here
\+ name: Commit Status Final

**Review by brlbil — CHANGES_REQUESTED** [review]

**Review by brlbil — APPROVED** [review]
LGTM

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+    name: 'Setup & Test'
+    env:
+      job_name: 'Setup & Test'
+    strategy:
+      fail-fast: false
+      max-parallel: 16
+      matrix:
+        include:
+          - name: '1'
+            kernel: '4.19-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '4.19-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'false'
+            tunnel: 'vxlan'
+
+          - name: '2'
+            kernel: '5.4-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.4-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'false'
+            tunnel: 'disabled'
+
+          - name: '3'
+            kernel: '5.10-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.10-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'false'
+            tunnel: 'disabled'
+            endpoint-routes: 'true'
+
+          - name: '4'
+            kernel: '5.10-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.10-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'true'
+            tunnel: 'vxlan'
+            lb-mode: 'snat'
+            endpoint-routes: 'true'
+            egress-gateway: 'true'
+
+          - name: '5'
+            kernel: '5.15-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.15-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'true'
+            tunnel: 'disabled'
+            lb-mode: 'dsr'
+            endpoint-routes: 'true'
+            egress-gateway: 'true'
+            host-fw: 'true'
+
+          - name: '6'
+            kernel: '6.0-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '6.0-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 9]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'none'
+            kpr: 'true'
+            tunnel: 'vxlan'
+            lb-mode: 'snat'
+            egress-gateway: 'true'
+            host-fw: 'true'
+            lb-acceleration: 'testing-only'
+
+          - name: '7'
+            kernel: 'bpf-next-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: 'bpf-next-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 10]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'none'
+            kpr: 'true'
+            tunnel: 'disabled'
+            lb-mode: 'snat'
+            egress-gateway: 'true'
+            lb-acceleration: 'testing-only'
+
+          - name: '8'
+            kernel: 'bpf-next-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: 'bpf-next-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 11]
```diff
@@ -0,0 +1,429 @@
...
+            kube-proxy: 'iptables'
+            kpr: 'false'
+            tunnel: 'geneve'
+            endpoint-routes: 'true'
+
+          - name: '9'
+            kernel: '5.10-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.10-20230810.091425'
```

File: .github/workflows/tests-e2e-upgrade.yaml [hunk 12]
```diff
@@ -0,0 +1,429 @@
...
+            kpr: 'true'
+            tunnel: 'vxlan'
+            encryption: 'wireguard'
+            encryption-node: 'false'
+            lb-mode: 'snat'
+            endpoint-routes: 'true'
+            egress-gateway: 'true'
+
+          - name: '10'
+            kernel: '5.15-20230420.212204'
```
Comment (aanm) [inline .github/workflows/tests-e2e-upgrade.yaml]:
```suggestion
            # renovate: datasource=docker depName=quay.io/lvh-images/kind
            kernel: '5.15-20230810.091425'
```

**Review by aanm — CHANGES_REQUESTED** [review]
Also, wouldn't it make more sense to add this as a step of the existing conformance-e2e.yaml?

**Comment by brb** [comment]
> Also, wouldn't it make more sense to add this as a step of the existing conformance-e2e.yaml?

@aanm This would prolong the ci-e2e exec time, which is currently ~15min.

**Review by aanm — CHANGES_REQUESTED** [review]
Why do we have `(${{ matrix.name }})` in all step names? Can we remove them?

**Comment by aanm** [comment]
> > Also, wouldn't it make more sense to add this as a step of the existing conformance-e2e.yaml?
> 
> @aanm This would prolong the ci-e2e exec time, which is currently ~15min.

but from what I can see the ci-e2e-upgrade is duplicating the same test done in the conformance-e2e. Maybe we should drop conformance-e2e because one of the steps done in the ci-e2e-upgrade has the same coverage as the conformance-e2e? What I mean is that the step ["Test Cilium after upgrade"](https://github.com/cilium/cilium/actions/runs/6047140499/job/16411576256#step:17:1) in `ci-upgrade` is doing the same thing as ["Run tests"](https://github.com/cilium/cilium/actions/runs/6047139929/job/16410019204#step:13:1) in `ci-e2e`

**Comment by brb** [comment]
> Maybe we should drop conformance-e2e because one of the steps done in the ci-e2e-upgrade has the same coverage as the conformance-e2e?

@aanm Unfortunately noup, as that step which is run by ci-e2e-upgrade might be influenced by a previous installation of a stable release.

> Why do we have (${{ matrix.name }}) in all step names? Can we remove them?

Can we distinguish in the dashboard which configuration failed w/o it?

**Review by aanm — APPROVED** [review]
For future reference. Before marking this tests as required we should make sure this is done https://github.com/cilium/cilium-cli/issues/1735#issuecomment-1710229091

**Comment by brb** [comment]
Let's wait with merging until https://github.com/cilium/cilium-cli/pull/1952 has been released.

**Comment by thorn3r** [comment]
cleanup: removing from GH projects and `needs-backport` labels

**Pull request closed without merging**